third_party/kubetest2/README.md

```diff
@@ -0,0 +1,208 @@
+
+```
+I1211 08:56:43.836568   14380 app.go:61] The files in RunDir shall not be part of Artifacts
+I1211 08:56:43.837144   14380 app.go:62] pass rundir-in-artifacts flag True for RunDir to be part of Artifacts
+I1211 08:56:43.837188   14380 app.go:64] RunDir for this run: "/home/ubuntu/go/e2e-framework/examples/_rundir/cf42e94d-1c2e-4c6f-9912-2ff948608008"
+I1211 08:56:43.860870   14380 app.go:128] ID for this run: "cf42e94d-1c2e-4c6f-9912-2ff948608008"
+Running:  go test  ./simple -args
+ok  	sigs.k8s.io/e2e-framework/examples/simple	0.363s
+```
+
+You can easily change this command to stand up, and teardown, a Kubernetes cluster using KinD for instance (`kubetest2` also supports other deployers):
+
+```
+$> kubetest2 kind --up --down --test=e2e-framework -- --packages .
+I1211 09:20:46.640997   19579 up.go:62] Up(): creating kind cluster...
+Creating cluster "kind" ...
+ • Ensuring node image (kindest/node:v1.25.3) 🖼  ...
+ ✓ Ensuring node image (kindest/node:v1.25.3) 🖼
+ • Preparing nodes 📦   ...
+...
+Running:  go test   -args
+PASS
+ok  	sigs.k8s.io/e2e-framework/kubetest2test	0.307s
+I1211 09:23:14.406049   19579 down.go:32] Down(): deleting kind cluster...
+Deleting cluster "" ...
+```
+
+> It should be noted that e2e-framework offers its own programmatic hooks for more flexible controls of pre- and post-test tasks, such as creating and tearing down named clusters (see the examples folder).
+
+### Passing arguments to e2e-framework tests
+The kubetest2 e2e-framework tester supports the ability to pass CLI arguments to your tests to control settings such as test execution filters and test behaviors. Recall, in order for your e2e-framework tests to receive CLI arguments, you must programmatically create the test environment to receive its configuration from the command-line:
+
+```go
+
+var tenv env.Environment
+
+func TestMain(m *testing.M) {
+	cfg, err := envconf.NewFromFlags() // important: read config from CLI flags
+	if err != nil {
+		log.Fatalf("failed to build config from flags: %s", err)
+	}
+	tenv = env.NewWithConfig(cfg)
+	os.Exit(tenv.Run(m))
+}
+
+func TestClusterObjects(t *testing.T) {
+	f := features.New("cluster-test")
+	f.Assess("pod-count", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+		var pods corev1.PodList
+		client, err := cfg.NewClient()
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = client.Resources("kube-system").List(context.TODO(), &pods)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if len(pods.Items) == 0 {
+			t.Fatal("no pods in namespace kube-system")
+		}
+		return ctx
+	})
+	f.Assess("dep-count", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+		var deps appsv1.DeploymentList
+		client, err := cfg.NewClient()
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = client.Resources().List(context.TODO(), &deps)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if len(deps.Items) == 0 {
+			t.Fatal("apps found")
```

Comment:
Not sure if this error is right. Why does this say `apps found` where the check is `== 0` ?